### DISCUSSION TITLE
Translation of some components of the SDK

### DISCUSSION THREAD

**Discussion by Nicolas-Schuhl** [monkvision/monkjs#318] [Ideas]
**Pain points:**

- Currently the Capture Component is only in English some developers require the possibility to have the capture component in french

**Useful:** 

- Existing mappings for parts, damages et vehicule details: [https://docs.google.com/spreadsheets/d/1xesvhBEPBOkOOHYluGgqeEKFn-E-4XDH64BeekmbJxg/edit?usp=sharing](https://docs.google.com/spreadsheets/d/1xesvhBEPBOkOOHYluGgqeEKFn-E-4XDH64BeekmbJxg/edit?usp=sharing)

**Comment by natsukiai** [comment]
### Solution
Implement our own translation context or use a package like `i18n`
* Create a Context and a Provider called `Translations`
* Add a 'language: string' and a `t: function` to the context
* Create a hook called `useTranslations` returning `t` and `language`
* In `t` function get and memoized the translation json file according to its language dep
* Format the first parameter string
* Maybe use a package for the format function to handle gender and plurality for example

### Time estimation: 1w
* **Research**: 1d
* **Code**: 2d
* **Review**: 1d
* **Fix**: 1d
